all/native/utils/Log.cpp

```diff
@@ -39,6 +39,15 @@ namespace carto {
         OutputDebugStringA("\n");
     }
 #endif
+#ifdef __EMSCRIPTEN__
+    enum LogType { LOG_TYPE_FATAL = 0, LOG_TYPE_ERROR = 1, LOG_TYPE_WARNING = 2, LOG_TYPE_INFO = 3, LOG_TYPE_DEBUG = 4 };
+
+    static void OutputLog(LogType logType, const std::string& tag, const char* text) {
+        if ((int)logType <= 1) {
+            printf("LogType: %d, tag: %s, message: %s\n", logType, tag.c_str(), text);
```

Comment:
I believe the 'if' around `printf` should be removed here, logging of specific message types can be enabled via `Log.setShowWarn`, `Log.setShowError`, ... and the disabled messages never reach this code.

Reply:
I used `emscripten_log` instead of `printf` and it covers all log types.